### ISSUE TITLE
Composer Autoloader

### ISSUE THREAD

**Issue by staabm** [composer/composer#1699]
composers generated autoloader looks like 

``` php
    public static function loadClassLoader($class)
    {
        if ('Composer\Autoload\ClassLoader' === $class) {
            require __DIR__ . '/ClassLoader.php';
        }
    }

    public static function getLoader()
    {
        if (null !== self::$loader) {
            return self::$loader;
        }

        spl_autoload_register(array('ComposerAutoloaderInit2da35074ce058f3466ead2485a734be8', 'loadClassLoader'));
        self::$loader = $loader = new \Composer\Autoload\ClassLoader();
        spl_autoload_unregister(array('ComposerAutoloaderInit2da35074ce058f3466ead2485a734be8', 'loadClassLoader'));

```

As you can see this autoloader relies on autoloading of "\Composer\Autoload\ClassLoader()" itself.

I noticed some performance problems in a legacy app in which we added composer  in parallel to a custom classloader.
Since our app registers its own classloader before including the composers classloader the invoked class-loading in this lines triggers our own classloader which unsuccessfully tries to load composers ClassLoader file.

Long story short: why do you rely on class-loading at all here instead of just requiring composers' loader?
